testing/upload_validation.ts

```diff
@@ -1194,12 +1202,19 @@ const genericObjectSchema: z.ZodTypeAny = z.lazy(() =>
 
 const objectPropertyUnionSchema = z
   .union([booleanPropertySchema, numberPropertySchema, stringPropertySchema, arrayPropertySchema, genericObjectSchema])
-  .refine(
-    // BaseSchema isn't exported, so Pick<BooleanSchema | EmailSchema, ...> here is an approximation.
-    (baseSchema: Pick<BooleanSchema | EmailSchema, 'mutable' | 'autocomplete' | 'codaType'>) =>
-      baseSchema.codaType === ValueHintType.Email || !baseSchema?.autocomplete || baseSchema.mutable,
-    `"mutable" must be true to set "autocomplete"`,
-  );
+  .refine((schema: Schema) => {
+    const schemaForAutocomplete = maybeUnwrapArraySchema(schema);
+    return schemaForAutocomplete?.codaType !== ValueHintType.SelectList || 'autocomplete' in schemaForAutocomplete;
+  }, 'You must set "codaType" to ValueHintType.SelectList when setting an "autocomplete" property.')
+  .refine((schema: Schema) => {
+    const schemaForAutocomplete = maybeUnwrapArraySchema(schema);
+    return (
+      schemaForAutocomplete?.codaType !== ValueHintType.SelectList ||
+      !schemaForAutocomplete?.autocomplete ||
+      // NOTE: This is intentionally schema.mutable rather than schemaForAutocomplete.mtuable.
+      schema.mutable
+    );
+  }, `"mutable" must be true to set "autocomplete"`);
```

Comment:
this error message may need another copy.

Reply:
A follow-up PR will both rename "autocomplete" to "options" and also remove this restriction entirely since we want people to be able to set up options for view-only filters.

I updated the check here instead of removing it just to avoid scope creep in this PR